lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/PoisonTest.cs

```diff
@@ -87,45 +88,48 @@ public void OnNext(IDriverStarted value)
 
             public void OnNext(IAllocatedEvaluator value)
```

Comment:
Note that the test doesn't crash a Task per se, instead it crashes the `Evaluator` because it triggers an uncaught Exception in Alarm. To add to our coverage, I think what we need to do here is to test for the list of FailedContexts instead of injecting the `PoisonEventHandler` in `ContextStart`. What do you think?